src/impl/wshandshake.hpp

```diff
@@ -57,6 +58,8 @@ class WsHandshake final {
 	string mHost;
 	string mPath;
 	std::vector<string> mProtocols;
+	std::map<string, string> mCustomHeaders;
+	std::multimap<string, string> mRequestHeaders;
```

Comment:
Is there a reason to use two different maps here? You could use only one multimap.

Reply:
Hey ! It's to add support for the websocket server. 

But good suggestion. Rather than custom and request we can just rename them to request headers or just headers up to you.